modular_splurt/code/game/objects/items/lewd_items/genital_equipment/chastity/chastity_cage.dm

```diff
@@ -0,0 +1,149 @@
+#define blacklisted_genitals list(/obj/item/organ/genital/breasts, /obj/item/organ/genital/belly)
+
+/obj/item/organ/genital/proc/can_be_chastened()
+	for(var/blacklisted_genital in blacklisted_genitals)
+		if(type in typesof(blacklisted_genital))
+			return FALSE
+	
+	return TRUE
+
+/obj/item/key/chastity_key
+	name = "cage key"
+	icon_state = "key"
+	desc = "You better not lose this."
+
+/obj/item/genital_equipment/chastity_cage
+	name = "chastity cage"
+	desc = "Feeling submissive yet?"
+	icon = 'modular_splurt/icons/obj/lewd_items/chastity.dmi'
+	icon_state = "standard_cage"
+	w_class = WEIGHT_CLASS_TINY
+
+	var/obj/item/key/chastity_key/key
+
+	var/break_require = TOOL_WIRECUTTER //Which tool is required to break the chastity_cage
+	var/break_time = 25 SECONDS
+
+	var/obj/item/organ/genital/genital
+	var/mutable_appearance/cage_overlay
+	var/worn_icon_state
+
+	var/overlay_layer = -(GENITALS_FRONT_LAYER - 0.01)
+	var/is_overlay_on
```

Comment:
Overall... how does the system for putting on/taking off/breaking out of chastity items work?

Also, it probably already does but to make sure all the items check for the chastity pref before being able to interact with a character

Reply:
To take it off you need its key or remote. To break it, it depends; wirecutters for the standard cage, welding tool for the metal and a multitool for the e-stim

And I'm pretty sure they do check for the pref